### PR TITLE
Validate trailing slash stripped on request build

### PR DESCRIPTION
**Pull request by eiri**
## PR summary

Add a test to validate that we are stripping a trailing slash on setting of service URL. The test for env file is not necessary, since in go-core slash stripped at later stage during request build.

## PR Checklist

Please make sure that your PR fulfills the following requirements:

- [x] The commit message follows the
[Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type  

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [x] New tests
- [ ] Build/CI related changes
- [ ] Documentation content changes
- [ ] Other (please describe)

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: common/cloudant_base_test.go [hunk 1]
```diff
@@ -167,4 +167,18 @@ var _ = Describe(`Cloudant custom base service UT`, func() {
...
+			URL:           "https://cloudant.example/",
+			Authenticator: &core.NoAuthAuthenticator{},
+		})
+		Expect(cloudant).ToNot(BeNil())
+		Expect(err).To(BeNil())
+
+		builder := core.NewRequestBuilder(core.GET)
+		_, err = builder.ResolveRequestURL(cloudant.Options.URL, "/db", map[string]string{})
+		Expect(err).To(BeNil())
+		Expect(builder.URL.String()).To(Equal("https://cloudant.example/db"))
```
Comment (vmatyusGitHub) [inline common/cloudant_base_test.go:182]: It would be good to add an another case for `GetServerInformation` case:
```go
		_, err = builder.ResolveRequestURL(cloudant.Service.Options.URL, `/`, nil)
```

File: common/cloudant_base_test.go [hunk 2]
```diff
@@ -167,4 +167,18 @@ var _ = Describe(`Cloudant custom base service UT`, func() {
...
+	It("Validates URL trailing slash is stripped", func() {
+		cloudant, err := NewBaseService(&core.ServiceOptions{
+			URL:           "https://cloudant.example/",
+			Authenticator: &core.NoAuthAuthenticator{},
+		})
+		Expect(cloudant).ToNot(BeNil())
+		Expect(err).To(BeNil())
+
+		builder := core.NewRequestBuilder(core.GET)
+		_, err = builder.ResolveRequestURL(cloudant.Options.URL, "/db", map[string]string{})
```
Comment (vmatyusGitHub) [inline common/cloudant_base_test.go:180]: Optional:
What do you think about these solutions?
```suggestion
		_, err = builder.ResolveRequestURL(cloudant.Options.URL, "/db", nil)
```
OR
A path parameter check might be added.

File: common/cloudant_base_test.go (hunk 1 above)
Comment (eiri) [reply common/cloudant_base_test.go:182]: No, this wouldn't work, because `go-sdk-core` has [unusual logic](https://github.com/IBM/go-sdk-core/blob/main/v5/core/request_builder.go#L167-L181) to deal with slashes in url - if `serviceURL` has trailing slash it trims prefix slash from the `path` and if `serviceURL` doesn't have trailing slash, it adds leading slash to the `path` when it's missing. So your example will end-up _with_ trailing slash in `builder.URL`, because it'll be first removed from `path`, but then kept as suffix on `urlString`.

Basically while all other core sdks are trimming trailing slash from `serviceURL`, `go-sdk-core` is doing it differently and de-duplicating slashes while constructing URL instead, that's why using "/db" here is more visible, it is to confirm de-duplication, that we will not end up with "https://cloudant.example//db" when have both trailing slash on service url and leading slash on path.

File: common/cloudant_base_test.go (hunk 2 above)
Comment (eiri) [reply common/cloudant_base_test.go:180]: Using `nil` is fine, we are not using builder for anything real, so no risk of runtime exception.

But I'm not sure what additional benefits path parameter check would give us, we already confirmed we have valid `url.URL`, because there is no error and `go-sdk-core` [validates URL for us](https://github.com/IBM/go-sdk-core/blob/main/v5/core/request_builder.go#L186-L189)

File: common/cloudant_base_test.go (hunk 1 above)
Comment (vmatyusGitHub) [reply common/cloudant_base_test.go:182]: Thanks for the explanation. Now I get this.

**Review by vmatyusGitHub — APPROVED** [review]
LGTM :+1:

**Review by ricellis — APPROVED** [review]
